pkg/operator/apiserver/controller/apiservice/apigroup.go

```diff
@@ -25,11 +25,14 @@ func newEndpointPrecondition(kubeInformers kubeinformers.SharedInformerFactory)
 		name      string
 	}
 
-	return func(apiServices []*apiregistrationv1.APIService) (bool, error) {
+	return func(apiServices []ManagedAPIService) (bool, error) {
 
 		coordinates := []coordinate{}
-		for _, apiService := range apiServices {
-			curr := coordinate{namespace: apiService.Spec.Service.Namespace, name: apiService.Spec.Service.Name}
+		for _, managedAPIService := range apiServices {
+			if !managedAPIService.Enabled {
+				continue
```

Comment:
since we are skipping disabled services I would simply return only enabled services in `GetAPIServicesToMangeFunc`

Reply:
Both `newEndpointPrecondition` and `checkDiscoveryForByAPIServices` can be probably given only a list of enabled services.